test/test_output_formatter.sh

```
@@ -0,0 +1,421 @@
+  printf '    ;;\n'                             >> "$mock_dir/gh"
+  printf '  *"logs"*)\n'                        >> "$mock_dir/gh"
+  printf "    printf '%%s' '%s'\n" "$log_content" >> "$mock_dir/gh"
+  printf '    ;;\n'                             >> "$mock_dir/gh"
+  printf '  *) exit 1 ;;\n'                    >> "$mock_dir/gh"
+  printf 'esac\n'                              >> "$mock_dir/gh"
+  chmod +x "$mock_dir/gh"
+}
+
+# run_script runs the target script with the mock directory prepended to PATH so the mock executables override real tools.
+run_script() {
+  PATH="$mock_dir:$ORIGINAL_PATH" bash "$script" "$@"
```

Shellcheck (0.11.0)

[warning] script is referenced but not assigned.

(SC2154)